src/realm/util/file.cpp

```diff
@@ -1653,6 +1654,42 @@ std::string File::resolve(const std::string& path, const std::string& base_dir)
 #endif
 }
 
+std::string File::parent_dir(const std::string& path)
+{
+#if REALM_HAVE_STD_FILESYSTEM
+    namespace fs = std::filesystem;
+    return fs::path(path).parent_path().string(); // Throws
+#else
+    auto is_sep = [](char c) -> bool {
+        return c == '/' || c == '\\';
+    };
+    std::vector<size_t> sep_indices;
```

Comment:
The code below seems a bit convoluted. Why not just
```
auto it = std::find_if(path.rbegin(), path.rend(), is_sep);
while (it != path.rend() && is_sep(*it))
    ++it;
return path.substr(0, path.rend() - it);
```

Reply:
'/' cannot be part of a multibyte value. All multibyte characters have the upper bit set to 1.

Reply:
Oh thanks, good catch. In that case I'll simplify it to your suggestion.